lib/compile/index.js

```diff
@@ -0,0 +1,96 @@
+import most from "most";
+import escodegen from "escodegen";
+
+import applyPlugins from "./apply-plugins";
+import construct from "./construct";
+import compileModules from "./modules/compile";
+import bootstrapBundles from "./bundles/bootstrap";
+import hashBundle from "./bundles/hash";
+import interpolateFilename from "./bundles/interpolate-filename";
+import dedupeExplicit from "./bundles/dedupe-explicit";
+import dedupeImplicit from "./bundles/dedupe-implicit";
+
+export default function compile (opts, compilation) {
+  compilation = compilation || {
```

Comment:
The name compilation is a bit hard to understand as it's not a noun/noun-phrase that I can infer things about without looking at more code. The only sentences I can think of that use compilation use it as kindof an adjective. That said, I see what you were trying to do and I may just be out of the loop on naming conventions for streaming code. I'm mostly used to variables that describe the final and not the transitional state of an object.